applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1062,6 +1205,34 @@ namespace Kratos
     //************************************************************************************
     //************************************************************************************
     
+    Matrix BaseSolidElement::CalculateDeltaDisplacement()
```

Comment:
no, if you do

Matrix f()
{
       Matrix a(3,3)
       return a;
}

and you call it as

Matrix Acalled = f();

it may well do a copy (it is also possible that the optimizer manages to optimize away the copy, however you can not be sure about that)